addons/mail/tests/test_mail_activity.py

```diff
@@ -1,10 +1,101 @@
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Done To-Do 1',
+            'date_deadline': '2025-09-02',
+        })
+        done_activity1.action_done()
+
+        done_activity2 = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('res.partner'),
+            'res_id': self.partner_admin.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Done To-Do 2',
+            'date_deadline': '2025-09-03'
+        })
+        done_activity2.action_done()
+
+        self.assertEqual(self.env['mail.activity'].search_count([]), 1)
+        self.assertEqual(self.env['mail.activity'].search_count([('date_done', '=', fields.Date.today())]), 2)
+        self.assertEqual(self.env['mail.activity'].search_count([
+            '&',
+            ('date_done', '=', fields.Date.today()),
+            ('date_deadline', '=', '2025-09-03'),
+        ]), 1)
+
+    def test_search_date_done_related_field(self):
+        """ Test searching through a related field (res.partner's activity_ids). """
+        partner_with_active_activity = self.env['res.partner'].create({'name': 'Test Partner1 for Activities'})
+        partner_with_done_activity = self.env['res.partner'].create({'name': 'Test Partner2 for Activities'})
+
+        self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('res.partner'),
+            'res_id': partner_with_active_activity.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Partner Active To-Do',
+            'date_deadline': '2025-09-06',
+        })
+
+        done_activity = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('res.partner'),
+            'res_id': partner_with_done_activity.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Partner Done To-Do',
+            'date_deadline': '2025-09-07',
+        })
+        done_activity.action_done()
+
+        self.assertEqual(self.env['res.partner'].search([
+            ('activity_ids.date_deadline', '=', '2025-09-06')
+        ]), partner_with_active_activity)
+
+        self.assertEqual(self.env['res.partner'].search([
+            ('activity_ids.date_done', '=', fields.Date.today()),
+            ]), partner_with_done_activity)
+
+        self.assertEqual(self.env['res.partner'].search([(
+            'activity_ids', 'any', [
+                '&',
+                ('date_done', '=', fields.Date.today()),
+                ('date_deadline', '=', '2025-09-07'),
+            ],
+        )]), partner_with_done_activity)
```

Comment:
Both tests look very similar, I would do both in the same unit test.

Or at least you can put the set-up inside setUpClass so that there's less common work being done